sympy/solvers/solveset.py

```diff
@@ -416,7 +419,12 @@ def _solve_trig(f, symbol, domain):
     elif solns is S.EmptySet:
         return S.EmptySet
     else:
-        return ConditionSet(symbol, Eq(f_original, 0), S.Reals)
+        #try solving through solve_decomposition
+        try:
+            result = solve_decomposition(f_original, symbol, domain)
```

Comment:
`solve_decomposition ` can make solveset work easy. Isn't it good if `transolve` got the simple Transcendental equation to solve from the `solve_decomposition` ?